mathics/core/expression.py

```diff
@@ -1103,6 +1113,9 @@ def replace_vars(self, vars, options=None,
                     leaves = [Expression('List', *func_params), body] + \
                         self.leaves[2:]
 
+        if len(vars) == 0:  # might just be a symbol set via Set[] we looked up here
```

Comment:
Good point, I wasn't even aware that `not` could be used in this way on a dict here. Changed that.